### PR TITLE
refactor(core): Extract trigger context out of NodeExecutionFunctions (no-changelog)

### PR DESCRIPTION
**Pull request by netroy**
## Summary
This PR extracts Trigger context object out of NodeExecutionFunctions, and adds unit tests for most of this code.

## Related Linear tickets, Github issues, and Community forum posts

CAT-277

## Review / Merge checklist

- [x] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) 
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [x] Tests included. 
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

### CODE REVIEW

File: packages/core/src/NodeExecuteFunctions.ts [hunk 1]
```diff
@@ -3346,14 +3344,6 @@ const getSSHTunnelFunctions = (): SSHTunnelFunctions => ({
 		await Container.get(SSHClientsManager).getClient(credentials),
 });
 
-const getSchedulingFunctions = (workflow: Workflow): SchedulingFunctions => {
```
Comment (tomi) [inline packages/core/src/NodeExecuteFunctions.ts:3349]: This was not used?

**Review by tomi — DISMISSED** [review]
🚀

File: packages/core/src/NodeExecuteFunctions.ts (hunk 1 above)
Comment (netroy) [reply packages/core/src/NodeExecuteFunctions.ts:3345]: this was only used in poll and trigger contexts. with those two extracted, this function isn't needed here. This code now lives in `helpers/scheduling-helpers` now.

**Review by tomi — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.67.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.67.0)

**Pull request merged**